rbi/stdlib/matrix.rbi

```diff
@@ -0,0 +1,577 @@
+
+  def elements_to_f(); end
+
+  def elements_to_i(); end
+
+  def elements_to_r(); end
+
+  def eql?(other); end
+
+  # Returns a hash-code for the vector.
+  def hash(); end
+
+  # Returns `true` iff all of vectors are linearly independent.
+  #
+  # ```
+  # Vector[1,0].independent?(Vector[0,1])
+  #   => true
+  #
+  # Vector[1,2].independent?(Vector[2,4])
+  #   => false
+  # ```
+  def independent?(*vs); end
+
+  # Returns the inner product of this vector with the other.
+  #
+  # ```ruby
+  # Vector[4,7].inner_product Vector[10,1]  => 47
+  # ```
+  #
+  #
+  # Also aliased as:
+  # [`dot`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-dot)
+  def inner_product(v); end
+
+  # Overrides
+  # [`Object#inspect`](https://docs.ruby-lang.org/en/2.6.0/Object.html#method-i-inspect)
+  def inspect(); end
+
+  # Returns the modulus (Pythagorean distance) of the vector.
+  #
+  # ```
+  # Vector[5,8,2].r => 9.643650761
+  # ```
+  #
+  #
+  # Also aliased as:
+  # [`r`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-r),
+  # [`norm`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-norm)
+  def magnitude(); end
+
+  # Alias for:
+  # [`collect`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-collect)
+  def map(); end
+
+  # Like
+  # [`Vector#collect2`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-collect2),
+  # but returns a [`Vector`](https://docs.ruby-lang.org/en/2.6.0/Vector.html)
+  # instead of an [`Array`](https://docs.ruby-lang.org/en/2.6.0/Array.html).
+  def map2(v); end
+
+  # Alias for:
+  # [`magnitude`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-magnitude)
+  def norm(); end
+
+  # Returns a new vector with the same direction but with norm 1.
+  #
+  # ```
+  # v = Vector[5,8,2].normalize
+  # # => Vector[0.5184758473652127, 0.8295613557843402, 0.20739033894608505]
+  # v.norm => 1.0
+  # ```
+  def normalize(); end
+
+  # Alias for:
+  # [`magnitude`](https://docs.ruby-lang.org/en/2.6.0/Vector.html#method-i-magnitude)
+  def r(); end
+
+  # Returns a vector with entries rounded to the given precision (see
+  # [`Float#round`](https://docs.ruby-lang.org/en/2.6.0/Float.html#method-i-round))
+  def round(ndigits = 0); end
+
+  # Returns the number of elements in the vector.
+  def size(); end
+
+  # Returns the elements of the vector in an array.
+  def to_a(); end
+
+  # Return a single-column matrix from this vector
+  def to_matrix(); end
+
+  # Overrides
+  # [`Object#to_s`](https://docs.ruby-lang.org/en/2.6.0/Object.html#method-i-to_s)
+  def to_s(); end
+
+  # Returns `true` iff all elements are zero.
+  def zero?(); end
+end
+
+# The `Matrix` class represents a mathematical matrix. It provides methods for
+# creating matrices, operating on them arithmetically and algebraically, and
+# determining their mathematical properties such as trace, rank, inverse,
+# determinant, or eigensystem.
+class Matrix
+  include ::Enumerable
+
+  Elem = type_member(:out)
```

Comment:
I think this makes `Matrix` unusable in a typed file. 

- `T::Matrix` is not defined
- `def Matrix.[]` is defined by Ruby

Ex: in `sig { params(m: Matrix).void }` Sorbet-static complains that `Matrix` is a generic without type argument. While `Matrix[Integer]` satisfies the static part, it fails at Runtime.